### PR TITLE
Add analytics for login, logout, and matched

### PR DESCRIPTION
**Pull request by tak-hntlabs**

### CODE REVIEW

File: server/src/analytics.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+export interface AnalyticsEvent {
+  clientId: string;
+  category: string;
+  action: string;
+  label?: string;
+  value?: number;
+}
+
+export async function sendAnalytics(event: AnalyticsEvent, logInstance?: FastifyLoggerInstance): Promise<void> {
```
Comment (brianathere) [inline server/src/analytics.ts]: Please swap log into the first position so it's the same as other code.

File: server/src/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  const promise = new Promise<void>(resolve => {
```
Comment (brianathere) [inline server/src/analytics.ts]: rename const promise to be const request

File: server/src/analytics.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          resolve();
+        });
+
+        req.write(params.toString());
+        req.end();
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } catch (error: any) {
+        logError(`Cannot send analytics. Error: ${error.stack}`, logInstance);
+        // Ignore any exceptions
+        resolve();
```
Comment (brianathere) [inline server/src/analytics.ts:65]: reject(error)?

File: server/src/web_rtc_signaling_server.ts [hunk 4]
```diff
@@ -102,12 +102,28 @@ export class WebRTCSignalingServer {
   ): Promise<void> {
     await this.connectionStore.addConnection(log, walletAddress, connection);
     log.info(`addUserConnection walletAddress: ${walletAddress}}`);
+    await sendAnalytics(
```
Comment (brianathere) [inline server/src/web_rtc_signaling_server.ts]: await Promise.all([]) these two promises so they can run in parallel

File: server/src/web_rtc_signaling_server.ts [hunk 5]
```diff
@@ -102,12 +102,28 @@ export class WebRTCSignalingServer {
...
+      },
+      log,
+    );
   }
 
   public async removeUserConnection(log: FastifyLoggerInstance, walletAddress: string) {
     log.info(`removeUserConnection connection walletAddress: ${walletAddress}`);
     await this.connectionStore.removeFromWaitingList(log, walletAddress);
     await this.connectionStore.deleteConnection(log, walletAddress);
+    await sendAnalytics(
```
Comment (brianathere) [inline server/src/web_rtc_signaling_server.ts]: await Promise.all([]) these three promises so they can run in parallel

**Review by brianathere — COMMENTED** [review]
Approved with suggestions

File: server/src/analytics.ts (hunk 3 above)
Comment (tak-hntlabs) [reply server/src/analytics.ts:65]: I was thinking of ignoring all exceptions from sending analytics.

Comment (brianathere) [reply server/src/analytics.ts:65]: OK, then I guess this is a fine level to do it.

File: server/src/web_rtc_signaling_server.ts (hunk 4 above)
Comment (tak-hntlabs) [reply server/src/web_rtc_signaling_server.ts]: fixed

File: server/src/analytics.ts (hunk 2 above)
Comment (tak-hntlabs) [reply server/src/analytics.ts]: fixed

File: server/src/analytics.ts (hunk 1 above)
Comment (tak-hntlabs) [reply server/src/analytics.ts]: fixed

File: server/src/web_rtc_signaling_server.ts (hunk 5 above)
Comment (tak-hntlabs) [reply server/src/web_rtc_signaling_server.ts]: fixed

**Review by brianathere — APPROVED** [review]

**Pull request merged**